src/main/java/io/github/jhipster/config/apidoc/SwaggerAutoConfiguration.java

```diff
@@ -19,93 +19,103 @@
 import springfox.documentation.service.ApiInfo;
-import springfox.documentation.service.Contact;
 import springfox.documentation.spi.DocumentationType;
 import springfox.documentation.spring.web.plugins.Docket;
 import springfox.documentation.swagger2.annotations.EnableSwagger2;
 
+import javax.servlet.Servlet;
+import java.nio.ByteBuffer;
+import java.util.*;
+
+import static io.github.jhipster.config.JHipsterConstants.SPRING_PROFILE_SWAGGER;
+import static springfox.documentation.builders.PathSelectors.regex;
+
 /**
  * Springfox Swagger configuration.
- *
+ * <p>
  * Warning! When having a lot of REST endpoints, Springfox can become a performance issue. In that case, you can use a
  * specific Spring profile for this class, so that only front-end developers have access to the Swagger view.
  */
 @Configuration
-@ConditionalOnClass({ ApiInfo.class, BeanValidatorPluginsConfiguration.class })
+@ConditionalOnWebApplication
+@ConditionalOnClass({
+    ApiInfo.class,
+    BeanValidatorPluginsConfiguration.class,
+    Servlet.class,
+    DispatcherServlet.class
+})
+@Profile(SPRING_PROFILE_SWAGGER)
+@AutoConfigureAfter(JHipsterProperties.class)
 @EnableSwagger2
 @Import(BeanValidatorPluginsConfiguration.class)
-@Profile(SPRING_PROFILE_SWAGGER)
-public class SwaggerConfiguration {
+public class SwaggerAutoConfiguration {
 
     public static final String STARTING_MESSAGE = "Starting Swagger";
     public static final String STARTED_MESSAGE = "Started Swagger in {} ms";
-    public static final String MANAGEMENT_TITLE_SUFFIX = "management API";
+    public static final String MANAGEMENT_TITLE_SUFFIX = "Management API";
     public static final String MANAGEMENT_GROUP_NAME = "management";
     public static final String MANAGEMENT_DESCRIPTION = "Management endpoints documentation";
 
-    private final Logger log = LoggerFactory.getLogger(SwaggerConfiguration.class);
+    private final Logger log = LoggerFactory.getLogger(SwaggerAutoConfiguration.class);
 
     private final JHipsterProperties.Swagger properties;
 
-    public SwaggerConfiguration(JHipsterProperties jHipsterProperties) {
+    public SwaggerAutoConfiguration(JHipsterProperties jHipsterProperties) {
         this.properties = jHipsterProperties.getSwagger();
     }
 
+    /**
+     * JHipster Swagger Customizer
+     *
+     * @return the Swagger Customizer of JHipster
+     */
+    @Bean
+    public JHipsterSwaggerCustomizer jHipsterSwaggerCustomizer() {
+        return new JHipsterSwaggerCustomizer(properties);
+    }
+
     /**
      * Springfox configuration for the API Swagger docs.
      *
      * @return the Swagger Springfox configuration
      */
     @Bean
-    public Docket swaggerSpringfoxApiDocket() {
+    @ConditionalOnMissingBean(name = "swaggerSpringfoxApiDocket")
+    public Docket swaggerSpringfoxApiDocket(List<SwaggerCustomizer> swaggerCustomizers,
+                                            ObjectProvider<AlternateTypeRule[]> alternateTypeRulesProviders) {
```

Comment:
Inject directly a List of AlternateTypeRules

Reply:
Fixed